### DISCUSSION TITLE
How to keep user logged in similar to the Google UX?

### DISCUSSION THREAD

**Discussion by Nefcanto** [keycloak/keycloak#21899] [Q&A]
Google's UX is very good and friendly. You log in and as long as you are a normal person (doing something with your PC once every couple of days) you stay logged in.

In Keycloak it's not like that. Either you constantly work with the system or you're kicked out very soon after some idle time.

Is it possible to get Google's UX here?

I thought about these changes, but I did not do them because I'm not sure and this is the production instance:

* Increasing `Access Token Lifespan` for the realm from **Realm Settings** / **Tokens**, and changing it from its default 5 minutes to something like 60 * 24 * 30 minutes (30 days)
* Increating `SSO Session Idel` from **Realm Settings** / **Session**, from its default 30 minutes to like 30 days

Yet there are too many configurations and in spite of having the docs and mini-helps beside them, they are confusing to me.

How about these fields?

* Client Session Idle
* Offline Session Idle

So, my question in short is:

What configurations should I change to make Keycloak behave like Google in its login experience?

**Comment by ddurham2** [comment]
It's very much determined by the session timeout times (not the access token times since tokens are obtained from sessions so long as the session exists).  Idle times refers to whether it'll timeout early if the user hasn't interacted with the session in a while.  Max times expire sessions even if they _have_ been interacting. 

Maybe something like...

![image](https://github.com/keycloak/keycloak/assets/4805134/ff52a880-a14b-4301-b397-c2d2ff8671bd)
